src/js/S3Client.js

```diff
@@ -237,11 +237,12 @@ export default class S3Client {
         Delete: {
           Objects: objects,
         },
+        BypassGovernanceRetention: true,
       };
       return this.client.deleteObjects(params, (error, data) => {
         error ? reject(error) : resolve(data);
       });
-    }).then(() => this._deleteFolders(bucketName, folders));
+    }).then(data => this._deleteFolders(bucketName, folders, data));
```

Comment:
What if we try to delete some folders ? The current implementation wouldn't raise any error I think.
I would go for a different approach where the error handling is performed in the client wrapper instead of delegating it to the client consumer. 

```suggestion

        BypassGovernanceRetention: true,
      };
      return this.client.deleteObjects(params, (error, data) => {
        const hasAccessDeniedError =
          data &&
          data.Errors &&
          data.Errors.some(error => error.Code === 'AccessDenied');

        if (error || hasAccessDeniedError){
           reject(error || data);
        }
        
        resolve(data);
      });
    }).then(() => this._deleteFolders(bucketName, folders));
    }).then(() => this._deleteFolders(bucketName, folders));
```

And then we keep the old  _deleteFolders  implementation.